generic/MaplGenericComponent.F90

```diff
@@ -12,16 +12,56 @@ module mapl_MaplGenericComponent
    use mapl_MaplComponent
    use mapl_ErrorHandlingMod
    use pFlogger
+   use mapl_OpenMP_Support
+   use mapl_MaplGrid
+   !$ use omp_lib
    implicit none
    private
 
    public :: MaplGenericComponent
+   public :: get_grid
+
+   procedure(), pointer :: user_method1 => null()
```

Comment:
There are several places in the layer where we assume the gridded component has 2 run phases (this is only the first place where we have this assumption). We could easily lift this assumption by looping over phases from 1 to size(this%run_entry_points) and calling ESMF_GridCompSetEntryPoint (instead of set_services)

Reply:
I will try this.

Reply:
@atrayano Do you mind we look at this together?